Civi/Api4/Action/CiviCase/CiviCaseSaveTrait.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+
+namespace Civi\Api4\Action\CiviCase;
+
+/**
+ * @inheritDoc
+ */
+trait CiviCaseSaveTrait {
+
+  /**
+   * @inheritDoc
+   */
+  protected function writeObjects($cases) {
+    $cases = array_values($cases);
+    $result = parent::writeObjects($cases);
+
+    // If the case doesn't have an id, it's new & needs to be opened.
+    foreach ($cases as $idx => $case) {
+      if (empty($case['id'])) {
+        $this->openCase($case, $result[$idx]['id']);
+      }
+    }
```

Comment:
Need to `return $result;` from writeObjects().

Reply:
:facepalm: